hibernate-core/src/main/java/org/hibernate/type/spi/TypeConfiguration.java

```diff
@@ -209,6 +215,27 @@ public void sessionFactoryClosed(SessionFactory factory) {
 		// 		valid while the TypeConfiguration is scoped to SessionFactory
 	}
 
+	public String matchTypeContribution(Mapping mapping, int columnType) {
```

Comment:
Anytime you are calling this, the TypeConfiguration has access to the `Mapping` it needs - passing it is redundant and adds a requirement on a class we know it going away (`Mapping`).